File: OJ/LeetCode/src/1614 - Maximum Nesting Depth of the Parentheses [Stack].cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int MaxDepth(string s)
    {
        int depth = 0;
        Stack<char> list = new Stack<char>();
        foreach (var c in s)
        {
            if (c == '(') list.Push(c);
            else if (c == ')') list.Pop();
            depth = Math.Max(list.Count, depth);
        }
        return depth;
    }
}

File: OJ/LeetCode/src/1614 - Maximum Nesting Depth of the Parentheses [Stack].cpp
#include <stack>

using namespace std;

class Solution {
public:
    int maxDepth(string s) {
        int depth = 0;
        stack<char> list;
        for (auto c : s) {
            if (c == '(') list.push(c);
            else if (c == ')') list.pop();
            depth = max((int)list.size(), depth);
        }

        return depth;
    }
};